modxcms/revolution
# [Port to 3.x] Fix Various Issues with Form Panel Validation and Field Display - Part 2, TV Panel

_NOTE: This is part 2 of 6 of a revised submission of #15887._

### What does it do?

Ports enhancements to the TV form panel from PR #15146 to 3.x (including subsequent fixes by others). 
Changes include:
1. Updated panel JS to utilize new shared methods introduced in the main FormPanel via part 1 of this PR set (already merged).
2. Updated mysql schema validation rules for all Elements (fixes incorrect lookahead for space at the end of Element names [changed to negative lookbehind]).
3. Fix Element Create and Update processors to always use the correct data index for names (`name` or `templatename`).
4. Addressed a concern raised by @Ruslan-Aleev in #15854 : (1) fine-tuned css so the copy icon that appears on hover over the current TV's MODX tag is not clickable.
5. Enhanced readability of field descriptions by darkening slightly on hover.
6. A handful of minor lexicon updates.

### Why is it needed?

Makes needed UI and consistency improvements to the TV panel.

### How to test

1. From your terminal app, run `grunt build` from within the `_build/templates/default` folder and clear your browser cache.
2. Create and update a few TVs to verify behavior and appearance is as expected.

### Related issue(s)/PR(s)
Partly addresses #15854 
Partial port of #15146
